lib/ansible/modules/system/xfconf.py

```diff
@@ -152,61 +158,78 @@ def main():
     # Assign module values to dictionary values
     channel = module.params['channel']
     property = module.params['property']
-    value_type = module.params['value_type']
-    if module.params['value'].lower() == "true":
-        value = "true"
-    elif module.params['value'] == "false":
-        value = "false"
+    values = module.params['value']
+    value_types = module.params['value_type']
+    if values is not None and value_types is not None:
+        if len(values) != len(value_types):
+            module.fail_json(msg='Same number of "value" and "value_type" needed')
+        for i in range(len(values)):
+            if values[i].lower() == "true" or values[i].lower() == "false":
+                values[i] = values[i].lower()
+        for value_type in value_types:
+            if value_type not in ['int', 'bool', 'float', 'string']:
+                module.fail_json(msg='value_type %s is not supported'
+                                 % str(value_type))
     else:
-        value = module.params['value']
+        values = value_types = None
+    array = module.params['force_array'] or values is not None and len(values) > 1
 
     state = state_values[module.params['state']]
 
     # Initialize some variables for later
     change = False
-    new_value = ''
+    new_values = ''
 
     if state != "get":
-        if value is None or value == "":
+        if values is None or values[0] == "":
             module.fail_json(msg='State %s requires "value" to be set'
                              % str(state))
-        elif value_type is None or value_type == "":
+        elif value_types is None or value_types[0] == "":
             module.fail_json(msg='State %s requires "value_type" to be set'
                              % str(state))
 
     # Create a Xfconf preference
     xfconf = XfconfPreference(module,
                               channel,
                               property,
-                              value_type,
-                              value)
-    # Now we get the current value, if not found don't fail
-    dummy, current_value = xfconf.call("get", fail_onerr=False)
-
-    # Check if the current value equals the value we want to set.  If not, make
-    # a change
-    if current_value != value:
+                              value_types,
+                              values,
+                              array)
+    # Now we get the current values, if not found don't fail
+    dummy, current_values = xfconf.call("get", fail_onerr=False)
+
+    # Convert current_values to array format
+    if "Value is an array with" in current_values:
+        current_values = current_values.split("\n")
```

Comment:
Lines 203-205 could be replaced with:
```suggestion
        current_values = current_values.split("\n")[2:]
```